### PR TITLE
solved problem when using -u in a non interactive shell

### PR DESCRIPTION
**Pull request by BrownTruck**
PS1 was set as undeclared (like when using shell provisionning in vagrant)

---

_This was automatically migrated from pypa/virtualenv#893 to reparent it to the `master` branch. Please see original pull request for any previous discussion._

_Original Submitter: @allan-simon_

### CODE REVIEW

**Comment by joshuaspence** [comment]
+1 on this

**Comment by johnomics** [comment]
+1, would close https://github.com/pypa/virtualenv/issues/1029 and https://github.com/pypa/virtualenv/issues/991

**Comment by andersk** [comment]
Workaround: set `VIRTUAL_ENV_DISABLE_PROMPT=t`. (But you should really just merge this.)

File: virtualenv_embedded/activate.sh [hunk 1]
```diff
@@ -54,11 +54,11 @@ if ! [ -z "${PYTHONHOME+_}" ] ; then
...
 
 if [ -z "${VIRTUAL_ENV_DISABLE_PROMPT-}" ] ; then
-    _OLD_VIRTUAL_PS1="$PS1"
+    _OLD_VIRTUAL_PS1="${PS1-}"
     if [ "x__VIRTUAL_PROMPT__" != x ] ; then
-        PS1="__VIRTUAL_PROMPT__$PS1"
+        PS1="__VIRTUAL_PROMPT__${PS1-}"
     else
-        PS1="(`basename \"$VIRTUAL_ENV\"`) $PS1"
+        PS1="(`basename \"$VIRTUAL_ENV\"`) ${PS1-}"
```
Comment (andersk) [inline virtualenv_embedded/activate.sh]: You need to run `bin/rebuild-script.py` and commit `virtualenv.py` to make these changes take effect.

Comment (allan-simon) [inline virtualenv_embedded/activate.sh]: done :)

**Comment by alexjurkiewicz** [comment]
Is there anything I can help with to get this PR merged? The issue is affecting me too.

**Comment by ssbarnea** [comment]
Did virtualenv became unmaintained, based on the severity of this pull request I would have reasons to believe so.

**Comment by elena-xiong** [comment]
@gaborbernat I've re-updated this branch, is there now anything preventing merging it ? :)

**Comment by allan-simon** [comment]
@gaborbernat  I've re-updated this PR,   is there's now anything preventing merging it (and this way I can finally tell my workmate to remove all the  set +u  / source active / set -u  we have everywhere in the code we've written these last years :)

**Comment by allan-simon** [comment]
thanks a lot

**Comment by ssbarnea** [comment]
I would like to thanks everyone for fixing this long lasting bug which was forcing people to avoid using the recommended bash strict mode.

**Pull request merged**